templates/workloads/partials/cf/publish.yml

```diff
@@ -0,0 +1,33 @@
+{{- if .Publish}}
+{{- range $topic := .Publish.Topics}}
+{{logicalIDSafe $topic.Name}}SNSTopic:
+    Type: AWS::SNS::Topic
+    Properties:
+      TopicName: ${AWS::StackName}-{{$topic.Name}}
+  
+  {{- if $topic.AllowedWorkers}}
+  {{logicalIDSafe $topic.Name}}SNSTopicPolicy:
+    Type: AWS::SNS::TopicPolicy
+    DependsOn: {{$topic.Name}}SNSTopic
+    Properties:
+      Topics: 
+        - !Ref {{$topic.Name}}SNSTopic
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          {{- range $worker := .AllowedWorkers}}
+          - Effect: Allow
+            Principal:
+              Service: sqs.amazonaws.com
+            Action: 
+              - sns:Subscribe
+            Resource: !Sub 'arn:${AWS::Partition}:sqs:${AWS::Region}:${AWS::AccountId}:*'
+            Condition:
+              StringEquals:
+                'sqs:ResourceTag/copilot-application': !Sub '${AppName}'
+                'sqs:ResourceTag/copilot-environment': !Sub '${EnvName}'
+                'sqs:ResourceTag/copilot-service': !Sub {{$worker}}
```

Comment:
```suggestion
                'sqs:ResourceTag/copilot-service': {{$worker}}
```
This doesn't need a `!Sub`, i don't believe, but it does mean we should make sure to validate all worker names at deploy time.

Reply:
Got rid of sub; worker names are tags to I ensured they were valid svc names.

Reply:
"Will CloudFormation fail on creation in this circumstance if the topic policy doesn't allow the queue to create a subscription?" I don't believe this is a concern for publish, as publish doesn't make the subscriptions, it just allows subscriptions by allowing any subscriptions with the tags given here (which is the worker svc name). That validation should be on the subscribe end.